lib/puppet/functions/katello/build_dn.rb

```diff
@@ -9,11 +9,12 @@
 Puppet::Functions.create_function(:'katello::build_dn') do
   # @param options
   dispatch :build_dn do
-    param 'Array[Tuple[String[1], Optional[String[1]]]]', :options
+    param 'Array[Tuple[String[1], Optional[String]]]', :options
     return_type 'String'
   end
 
   def build_dn(options)
-    options.select { |_key, value| value }.map { |key, value| "#{key}=#{value}" }.join(', ')
+    options_with_values = options.filter { |_key, value| !value.nil? && value != '' }
```

Comment:
I think this is more expressive in what it tries to achieve:
```suggestion
    options_with_values = options.delete_if { |_key, value| value.nil? || value == '' }
```

I admit Ruby is always a bit weird since there's multiple ways of doing things. For example `filter` is an alias for `select`.

Reply:
I think if we use `delete_if` it becomes:

```
  options.delete_if { |_key, value| value.nil? || value == '' }
```

My understanding reading the docs on it is that it's an in place modifier.

Reply:
And testing, this actually leads to an error:

```
        expected katello::build_dn([["a", nil], ["b", "2"]]) to have returned "b=2" instead of raising FrozenError(can't modify frozen Array: [["a", nil], ["b", "2"]])
        /home/ehelms/workspace/upstream/installer/puppet-katello/spec/fixtures/modules/katello/lib/puppet/functions/katello/build_dn.rb:17:in `delete_if'
```

Reply:
That makes sense, thanks.

Though I think you should use `select` instead of `filter:
```
  Evaluation Error: Error while evaluating a Function Call, undefined method `filter' for #<Array:0x0000561f726d9560> (file: /home/runner/work/puppet-katello/puppet-katello/spec/fixtures/modules/katello/manifests/application.pp, line: 40, column: 24) on node fv-az572-767.xfljd2mj00zetb5gvhv1u2zrtd.dx.internal.cloudapp.net
```